src/main/kotlin/com/openlattice/assembler/processors/AddMembersToOrganizationAssemblyProcessor.kt

```diff
@@ -43,21 +42,18 @@ data class AddMembersToOrganizationAssemblyProcessor(
         ReadOnly {
 
     @Transient
-    private lateinit var acm: AssemblerConnectionManager
+    private var acm: AssemblerConnectionManager? = null
 
     override fun process(entry: MutableMap.MutableEntry<UUID, OrganizationAssembly?>): Void? {
         val organizationId = entry.key
         val assembly = entry.value
-        if (assembly == null) {
-            throw IllegalStateException("Encountered null assembly while trying to add new principals " +
-                    "${authorizedPropertyTypesOfEntitySetsByPrincipal.keys} to organization $organizationId.")
-        } else {
-            val dbName = PostgresDatabases.buildOrganizationDatabaseName(organizationId)
-            acm.connect(dbName)
-                    .use { dataSource ->
-                        acm.addMembersToOrganization(dbName, dataSource, authorizedPropertyTypesOfEntitySetsByPrincipal)
-                    }
-        }
+        assembly ?: throw IllegalStateException("Encountered null assembly while trying to add new principals " +
+                "${authorizedPropertyTypesOfEntitySetsByPrincipal.keys} to organization $organizationId.")
+
+        val dbName = PostgresDatabases.buildOrganizationDatabaseName(organizationId)
+        acm?.connect(dbName)?.let { dataSource ->
```

Comment:
What's the purpose of allow acm to be null?

Reply:
In case, the `init` function is not called on the processor I think it's more appropriate to throw an exception then to ignore it (and get e nullpointer later)

Reply:
Idiomatic way of doing this in kotlin is to leverage the compiler instead of using nullable type. ```check( ::acm.isInitialized ) { "ACM was not initialized before processing." }```

Reply:
https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/is-initialized.html

Reply:
Oh, cool! Will do!